src/controller/mod.rs

```diff
@@ -221,9 +217,13 @@ impl IntoResponse for Error {
                     ErrorDetail::with_reason("Bad Request"),
                 )
             }
-            _ => (
+            Self::BadRequest(err) => (
                 StatusCode::BAD_REQUEST,
-                ErrorDetail::with_reason("Bad Request"),
+                ErrorDetail::new("Bad Request", &err),
+            ),
+            _ => (
+                StatusCode::INTERNAL_SERVER_ERROR,
+                ErrorDetail::with_reason("Internal Server Error"),
```

Comment:
This is the original response
```
ErrorDetail::new("internal_server_error", "Internal Server Error"),
```

Why did you change it?

Reply:
I had to implement the bad request error which carries a reason, and then saw that internal server error had some sort of enum style error message and the same but formatted in the description, so figured it was probably prototyped and never fully implemented, so I added the change to this PR. Happy to submit it in another PR if you like.

Reply:
The 500 responses should follow this format:
```json
{
    "error":"internal_server_error",
    "description":"Internal Server Error"
}
```

instead of the format you changed it to:
```json
{
    "error":"Internal Server Error"
}
```

Reply:
But why the specific format though? The error message looks very arbitrary, the description contains the same as the error itself, where other error types don't have that.

Reply:
I believe all errors should follow this format, as it provides a clear, standardized error code for API responses.

Reply:
Cool, will update